cached_network_image_platform_interface/lib/cached_network_image_platform_interface.dart

```diff
@@ -21,37 +22,37 @@ enum ImageRenderMethodForWeb {
 class ImageLoader {
   /// loads the images async and gives the resulted codecs on a Stream. The
   /// Stream gives the option to show multiple images after each other.
-  @Deprecated('use loadBufferAsync instead')
-  Stream<ui.Codec> loadAsync(
+  @Deprecated('Use loadImageAsync instead')
+  Stream<ui.Codec> loadBufferAsync(
     String url,
     String? cacheKey,
     StreamController<ImageChunkEvent> chunkEvents,
-    DecoderCallback decode,
+    DecoderBufferCallback decode,
     BaseCacheManager cacheManager,
     int? maxHeight,
     int? maxWidth,
     Map<String, String>? headers,
-    Function()? errorListener,
+    ValueChanged<Object>? errorListener,
```

Comment:
Changing the interface of the errorListener from Fuction() to Function(Object) is a breaking change as current implementations will stop to work. We have to upgrade the major version and I'm wondering if there is any need to keep `loadBufferAsync` in that situation. I think we can just remove it completely from this interface.